src/PAModel/Serializers/SourceSerializer.cs

```diff
@@ -950,5 +953,39 @@ private static void AddTestSuiteControl(CanvasDocument app, BlockNode controlIR)
             }
             app._screens[AppTestControlName].Children.Add(controlIR);
         }
+
+        /// <summary>
+        /// If the control name length is longer than 50, it is truncated and appended with a hash (to avoid collisions).
+        /// Checks the length of the escaped name, since its possible that the length is under 60 before escaping but goes beyond 60 later.
+        /// We do modulo by 1000 of the hash to limit it to 3 characters.
+        /// </summary>
+        /// <returns></returns>
+        private static string TruncateControlNameIfTooLong(string controlName)
+        {
+            var newControlName = Utilities.EscapeFilename(controlName);
+            if (newControlName.Length > MaxControlNameLength)
+            {
+                // limit the hash to 3 characters by doing a module by 1000
+                var hash = (GetHash(newControlName) % 1000).ToString("x3");
```

Comment:
>1000 [](http://example.com/codeflow?start=54&length=4)

nit: Since you're printing as hex, could this be 16^3 (instead of 10^3)

Reply:
we are doing the modulo by 1000 before converting to hex, so the max value would 999 in decimal.

Reply:
999.ToString("x3") // "3e7"
(16*16*16-1).ToString("x3") // "fff"

Reply:
basically, if the goal is to condense the hash to 3 hex digits, I think you should replace '1000' with (16*16*16-1)